File: results/llm_outputs/codellama7b/GuruSamples/java/replace-method-with-method-object_refactored.java
class Order {
  //...
  public double price() {
    double primaryBasePrice = computeBasePrice(primary);
    double secondaryBasePrice = computeBasePrice(secondary);
    double tertiaryBasePrice = computeBasePrice(tertiary);
    // Perform long computation.
  }

  private double computeBasePrice(Product product) {
    // Perform long computation.
  }
}

File: results/llm_outputs/codellama7b/GuruSamples/typescript/replace-method-with-method-object_refactored.ts
class Order {
  //...
  price(): number {
    const primaryBasePrice = this.computeBasePrice();
    const secondaryBasePrice = this.computeBasePrice();
    const tertiaryBasePrice = this.computeBasePrice();
    // Perform long computation.
  }

  private computeBasePrice(): number {
    // Perform long computation.
  }
}
